### PR TITLE
Move `withContentLength` from Message to  Headers

### PR DESCRIPTION
**Pull request by bplommer**
Following discussion on #6282 - I think this is too low-level to expose on `Message`.

### CODE REVIEW

**Comment by armanbilge** [comment]
Definitely agree this is a low-level and kind of unsafe method. I guess I'm confused in what context this is used (it seems we don't use it internally?). In those cases, aren't you more likely interacting with a `Message`?

**Comment by bplommer** [comment]
It's used in `toStrict` - I think that's the only place.

File: core/shared/src/main/scala/org/http4s/Message.scala [hunk 1]
```diff
@@ -301,7 +253,7 @@ sealed trait Message[F[_]] extends Media[F] { self =>
   def toStrict(implicit F: Concurrent[F]): F[Self] =
     body.compile.to(Chunk).map { chunk =>
       withBodyStream(Stream.chunk(chunk))
-        .withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong))
+        .transformHeaders(_.withContentLength(`Content-Length`.unsafeFromLong(chunk.size.toLong)))
     }
```
Comment (armanbilge) [inline core/shared/src/main/scala/org/http4s/Message.scala:257]: Ah sorry, missed this. Hm. Also now I realize I'm not sure how this interacts with https://github.com/http4s/http4s/pull/6286 which places the `Content-Length` header on the entity itself. Shouldn't that be the canonical?

**Review by rossabaker — APPROVED** [review]
@armanbilge's concern is legitimate, but I am currently leaning that we ought to have a `Length` value type.  If people agree with that, then I think this is good as is.

File: core/shared/src/main/scala/org/http4s/Message.scala (hunk 1 above)
Comment (rossabaker) [reply core/shared/src/main/scala/org/http4s/Message.scala:257]: Well, that would be introduced on main, and doesn't change this for 0.23 purposes, right?  I don't think #6286 is a reason to hold this after all?

Comment (armanbilge) [reply core/shared/src/main/scala/org/http4s/Message.scala:257]: Ah, right. Didn't mean to hold this one up.

**Review by armanbilge — APPROVED** [review]

**Pull request merged**